doc/rbd/rbd-snapshot.rst

```diff
@@ -206,107 +232,131 @@ performed these steps, you can begin cloning the snapshot.
-   may clone the snapshots without the ability to write or execute within the pool.
-
-#. **Image Migration/Recovery:** One way to use block device layering is to migrate
-   or recover data from one pool into another pool.
+ID, the image ID, and the snapshot ID. The inclusion of the pool ID means that
+you may clone snapshots from one pool to images in another pool.
+
+#. **Image Template:** A common use case for block device layering is to create
+   a master image and a snapshot that serves as a template for clones. For
+   example: a user may create an image for a Linux distribution (for example,
+   Ubuntu 12.04) and create a snapshot for it. The user may occasionally update
+   the image and create a new snapshot (by using such commands as ``sudo
+   apt-get update``, ``sudo apt-get upgrade``, or ``sudo apt-get dist-upgrade``
+   followed by ``rbd snap create``). As the image matures, the user can clone
+   any one of the snapshots.
+
+#. **Extended Template:** A more advanced use case includes extending a
+   template image that provides more information than a base image. For
+   example, a user may clone an image (for example, a VM template) and install
+   other software (for example, a database, a content management system, an
+   analytics system) and then snapshot the extended image, which may itself be
+   updated just like the base image.
+
+#. **Template Pool:** One way to use block device layering is to create a pool
+   that contains both (1) master images that act as templates and (2) snapshots
+   of those templates. You may then extend read-only privileges to users so
+   that they may clone the snapshots without the ability to write or execute
+   within the pool.
+
+#. **Image Migration/Recovery:** One way to use block device layering is to
+   migrate or recover data from one pool into another pool.
 
 Protecting a Snapshot
 ---------------------
 
-Clones access the parent snapshots. All clones would break if a user inadvertently 
-deleted the parent snapshot. To prevent data loss, you **MUST** protect the
-snapshot before you can clone it. ::
+Clones access the parent snapshots. All clones would break if a user
+inadvertently deleted the parent snapshot. To prevent data loss, you **MUST**
+protect the snapshot before you can clone it:
+
+.. prompt:: bash $
 
-	rbd snap protect {pool-name}/{image-name}@{snapshot-name}
+   rbd snap protect {pool-name}/{image-name}@{snapshot-name}
 
-For example::
+For example:
 
-	rbd snap protect rbd/my-image@my-snapshot
+.. prompt:: bash $
+
+   rbd snap protect rbd/my-image@my-snapshot
 
 .. note:: You cannot delete a protected snapshot.
 
 Cloning a Snapshot
 ------------------
 
-To clone a snapshot, specify you need to specify the parent pool, image and
-snapshot; and, the child pool and image name. You must protect the snapshot
-before  you can clone it. ::
+To clone a snapshot, specify the parent pool, the image, and the snapshot; and
```

Comment:
Feel free to ignore, but "specify the parent pool, image and snapshot" read better than "specify the parent pool, the image, and the snapshot".  At least to me, a single "the" article made it clear that "parent" applies to all of pool, image and snapshot -- which is how it actually is.  With your wording, "parent" needs to be repeated twice more to make that clear.

Reply:
@idryomov: One of the areas that needs the most work in this text is the definite articles (and please don't get me wrong or take this as mordant criticism: the proper deployment of English articles is one of the slipperiest parts of English grammar). Is the following clause correct?

"To clone a snapshot, specify the parent pool, the parent image, and the parent snapshot;"

Reply:
> @idryomov: One of the areas that needs the most work in this text is the definite articles (and please don't get me wrong or take this as mordant criticism: the proper deployment of English articles is one of the slipperiest parts of English grammar).

It wasn't me who wrote this, I would have done a worse job ;)

> Is the following clause correct?
> 
> "To clone a snapshot, specify the parent pool, the parent image, and the parent snapshot;"

Yes, it's correct.

Reply:
> It wasn't me who wrote this, I would have done a worse job ;)

You sell yourself short. 
 
> > Is the following clause correct?
> > "To clone a snapshot, specify the parent pool, the parent image, and the parent snapshot;"
> 
> Yes, it's correct.

This has been added to the doc.